autoload/lsp/internal/work_done_progress.vim

```diff
@@ -0,0 +1,45 @@
+let s:lsp_progress = {
+      \ 'server': '',
+      \ 'token': '',
+      \ 'title': '',
+      \ 'messages': '',
+      \ 'percentage': 100,
+      \ }
+
+function! lsp#internal#work_done_progress#_enable() abort
+    if !g:lsp_work_done_progress_enabled | return | endif
+
+    let s:Dispose = lsp#callbag#pipe(
+          \ lsp#stream(),
+          \ lsp#callbag#filter({x->has_key(x, 'response') && has_key(x['response'], 'method')
+          \  && x['response']['method'] ==# '$/progress' && has_key(x['response'], 'params')
+          \  && has_key(x['response']['params'], 'value') && type(x['response']['params']['value']) == type({})}),
+          \  lsp#callbag#subscribe({'next': function('s:handle_work_done_progress')})
+          \ )
+endfunction
+
+function! s:handle_work_done_progress(progress) abort
```

Comment:
do we need `lsp_diagnostics_updated` autocmd?

let us not add it yet if we don't require for vim airline or lightline.

Reply:
Not necessary at this time.
But we may want it in the future.

Not implemented for now.

Reply:
Sorry.
Added `lsp_progress_updated` autocmd, which I said was unnecessary before.
So, I attempted several implementations that did not use autocmd.
But, the implementation made users wait since progress notification is very frequently.  

So I added it separately from lsp_diagnostics_updated to reduce the update frequency by the status line plugin side.

Reply:
i mean it should had been `lsp_progress_updated`. what you have is better. thanks.